src/AsciiManager.cpp

```diff
@@ -167,6 +167,80 @@ void AsciiManager::CutChain()
     // to free it!
 }
 
+#pragma var_order(padding_1, i, string, text, guiString, charWidth, padding_2, padding_3);
+void AsciiManager::DrawStrings(void)
+{
+    i32 padding_1;
+    i32 padding_2;
+    i32 padding_3;
+    i32 i;
+    BOOL guiString;
+    f32 charWidth;
+    AsciiManagerString *string;
+    u8 *text;
+
+    guiString = TRUE;
+    string = this->strings;
+    this->vm0.flags |= AnmVmFlags_0;
+    this->vm0.flags |= AnmVmFlags_9 | AnmVmFlags_8;
+    for (i = 0; i < this->numStrings; i++, string++)
+    {
+        this->vm0.pos = string->position;
+        text = (u8 *)string->text;
+        this->vm0.scaleX = string->scale.x;
+        this->vm0.scaleY = string->scale.y;
+        charWidth = 14 * string->scale.x;
+        if (guiString != string->isGui)
+        {
+            guiString = string->isGui;
+            if (guiString)
+            {
+                g_Supervisor.viewport.X = g_GameManager.arcadeRegionTopLeftPos.x;
+                g_Supervisor.viewport.Y = g_GameManager.arcadeRegionTopLeftPos.y;
+                g_Supervisor.viewport.Width = 640.0;
```

Comment:
This is wrong, it should be assigning g_GameManager.arcade_region_size.x according to the assembly. Same for the Height below.